### PR TITLE
add primitive to fetch content of private repos during e2e setup

### PR DESCRIPTION
**Pull request by xh3b4sd**
Used in https://github.com/giantswarm/aws-operator/pull/1644 to fetch the SSH UserList from the installations repo, which is then used to configure e2e tests properly.

### CODE REVIEW

File: privaterepo/private_repo.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	p := &PrivateRepo{
+		owner: config.Owner,
+		repo:  config.Repo,
+		token: config.Token,
+	}
+
+	return p, nil
+}
+
+func (p *PrivateRepo) Content(ctx context.Context, path string) (string, error) {
```
Comment (xh3b4sd) [inline privaterepo/private_repo.go:43]: That is just some very abstract way of getting the content of some file from Github.

File: privaterepo/ssh_user_list.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"bytes"
+	"html/template"
+	"net/url"
+	"time"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/valuemodifier/path"
+)
+
+func ContentToSSHUserList(content string) (string, error) {
```
Comment (xh3b4sd) [inline privaterepo/ssh_user_list.go:13]: This is a more opinionated way of extracting the SSH UserList from the default draughtsman config.

File: privaterepo/ssh_user_list.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	var err error
+
+	var rendered string
+	{
+		tmpl, err := template.New("main").Parse(content)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+
+		v := struct {
```
Comment (xh3b4sd) [inline privaterepo/ssh_user_list.go:23]: I copied that more or less from `opsctl` to just render the template. The input data does not matter as we only need the template rendered to be able to read the SSH UserList from the resulting YAML. I chose this way to not waste more time on revamping the whole thing with opsctl and the installations repo.

**Review by rossf7 — APPROVED** [review]
LGTM

File: privaterepo/ssh_user_list.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+			AWS struct {
+				Region      string
+				HostCluster struct {
+					Account          string
+					AdminRoleARN     string
```
Comment (MarcelMue) [inline privaterepo/ssh_user_list.go:28]: Is it okay to have information like this possibly in our build logs? We store it in a configmap but having it accidentally exposed in public build logs seems like a step above that.

**Review by MarcelMue — APPROVED** [review]
Overall LGTM

File: privaterepo/ssh_user_list.go (hunk 4 above)
Comment (xh3b4sd) [reply privaterepo/ssh_user_list.go:28]: I don't follow. This is not logged at all. This is only used to render the template so we can access the data we are interested in.

Comment (MarcelMue) [reply privaterepo/ssh_user_list.go:28]: It is not logged, I am just wondering how bad it would be if someone were to log it accidentally.

Comment (xh3b4sd) [reply privaterepo/ssh_user_list.go:28]: Buddy, this is only the data structure used to render the template. There is no value given for it. It is only an empty string.

Comment (MarcelMue) [reply privaterepo/ssh_user_list.go:28]: I see the error in my ways here. Still curious how carefully we should treat this data (when it's in eventually).

File: privaterepo/ssh_user_list.go (hunk 2 above)
Comment (kopiczko) [reply privaterepo/ssh_user_list.go:13]: It should ctx as a first arg.

Comment (xh3b4sd) [reply privaterepo/ssh_user_list.go:13]: But that does not make sense here. There is no long running or asynchronous or synchronized logic being executed. It is only string to string transformation.

**Pull request merged**